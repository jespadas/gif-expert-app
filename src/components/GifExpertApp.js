import React, { useState } from 'react';

import { AddCategory } from './AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    // Usando .concat
    /* const handleAdd = () => {
        setCategories(categories => categories.concat("New category " + Math.random(1)));
    } */

    // Usando Spread ...
    /* const handleAdd = () => {
        setCategories(categories => [...categories, "New category "]);
    } */

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory />
            <hr />



            <ul>
                {categories.map((category) => {
                    return <li key={category}>{category}</li>
                })}
            </ul>

        </>
    )
}