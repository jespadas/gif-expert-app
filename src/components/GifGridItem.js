import React from 'react'

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="animate__animated animate__zoomIn">
            <img
                alt={title}
                src={url}
            />
            <p>{title}</p>
        </div>
    )
}
