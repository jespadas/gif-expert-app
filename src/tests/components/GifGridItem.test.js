import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = "el titulo";

    const url = "https://localhost:3000/img.png";

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe tener un parrafo con un parrafo en el titulo', () => {

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    })

    test('Debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find(('img'));

        // Para ver las props
        //console.log(img.prop('src'));

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('Debe de tener las clases animate__animated animate__zoomIn', () => {

        const div = wrapper.find('div');

        const className = div.prop('className');

        expect(className.includes('animate__zoomIn')).toBe(true);
    })

})