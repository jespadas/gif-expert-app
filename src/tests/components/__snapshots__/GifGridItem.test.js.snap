// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="animate__animated animate__zoomIn"
>
  <img />
  <p />
</div>
`;
